aptos-move/framework/aptos-framework/sources/dispatchable_fungible_asset.move

```diff
@@ -0,0 +1,192 @@
+        let func_opt = fungible_asset::withdraw_dispatch_function(store);
+        if (option::is_some(&func_opt)) {
+            assert!(
+                features::dispatchable_fungible_asset_enabled(),
+                error::aborted(ENOT_ACTIVATED)
+            );
+            let start_balance = fungible_asset::balance(store);
+            let func = option::borrow(&func_opt);
+            function_info::load_module_from_function(func);
+            let fa = dispatchable_withdraw(
+                store,
+                amount,
+                borrow_transfer_ref(store),
+                func,
+            );
+            let end_balance = fungible_asset::balance(store);
+            assert!(amount <= start_balance - end_balance, error::aborted(EAMOUNT_MISMATCH));
+            fa
+        } else {
+            fungible_asset::withdraw_non_dispatch(owner, store, amount)
+        }
+    }
+
+    /// Deposit `amount` of the fungible asset to `store`.
+    ///
+    /// The semantics of deposit will be governed by the function specified in DispatchFunctionStore.
+    public fun deposit<T: key>(store: Object<T>, fa: FungibleAsset) acquires TransferRefStore {
+        let func_opt = fungible_asset::deposit_dispatch_function(store);
+        if (option::is_some(&func_opt)) {
+            assert!(
+                features::dispatchable_fungible_asset_enabled(),
+                error::aborted(ENOT_ACTIVATED)
+            );
+            let func = option::borrow(&func_opt);
+            function_info::load_module_from_function(func);
+            dispatchable_deposit(
+                store,
+                fa,
+                borrow_transfer_ref(store),
+                func
+            )
+        } else {
+            fungible_asset::deposit_non_dispatch(store, fa)
+        }
+    }
+
+    /// Transfer an `amount` of fungible asset from `from_store`, which should be owned by `sender`, to `receiver`.
+    /// Note: it does not move the underlying object.
+    public entry fun transfer<T: key>(
+        sender: &signer,
+        from: Object<T>,
+        to: Object<T>,
+        amount: u64,
+    ) acquires TransferRefStore {
+        let fa = withdraw(sender, from, amount);
+        deposit(to, fa);
+    }
+
+    /// Transfer an `amount` of fungible asset from `from_store`, which should be owned by `sender`, to `receiver`.
+    /// The recipient is guranteed to receive asset greater than the expected amount.
+    /// Note: it does not move the underlying object.
+    public entry fun transfer_assert_minimum_deposit<T: key>(
+        sender: &signer,
+        from: Object<T>,
+        to: Object<T>,
+        amount: u64,
+        expected: u64
+    ) acquires TransferRefStore {
+        let start = fungible_asset::balance(to);
+        let fa = withdraw(sender, from, amount);
+        deposit(to, fa);
+        let end = fungible_asset::balance(to);
+        assert!(end - start >= expected, error::aborted(EAMOUNT_MISMATCH));
+    }
```

Comment:
@alnoki , we will move forward with this in this release. Please advise if we should add more in future releases.

Reply:
@davidiw thanks for the tag

As I mentioned on the AIP discussion ( https://github.com/aptos-foundation/AIPs/issues/374#issuecomment-2077648989 )

Is there anyway to also have `transfer_assert_fixed_deposit` ?